stepfunctions-httpendpoint-sam/template.yaml

```diff
@@ -0,0 +1,107 @@
+#  STEP FUNCTION LOG GROUP                                               #
+##########################################################################
+  StateMachineLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Join [ "/", [ "stepfunctions", StateMachineHTTPEndpoint]]
+       
+##########################################################################
+#   Roles                                                               #
+##########################################################################
+  StatesExecutionRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - !Sub states.${AWS::Region}.amazonaws.com
+            Action: "sts:AssumeRole"
+      Path: "/"
+      Policies:
+        - PolicyName: CWLogs
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogDelivery"
+                  - "logs:CreateLogStream"
+                  - "logs:GetLogDelivery"
+                  - "logs:UpdateLogDelivery"
+                  - "logs:DeleteLogDelivery"
+                  - "logs:ListLogDeliveries"
+                  - "logs:PutLogEvents"
+                  - "logs:PutResourcePolicy"
+                  - "logs:DescribeResourcePolicies"
+                  - "logs:DescribeLogGroups"
+                Resource: "*"
+        - PolicyName: StatesExecutionPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "events:RetrieveConnectionCredentials"
+                Resource:
+                  - !GetAtt EventBridgeConnection.Arn
+              - Effect: Allow
+                Action:
+                  - "secretsmanager:GetSecretValue"
+                  - "secretsmanager:DescribeSecret"
+                Resource:
+                  - !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:events!connection/*"
+              - Effect: Allow
+                Action:
+                  - "states:InvokeHTTPEndpoint"
+                Resource:
+                  - !Sub "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:*"           
+
+##########################################################################
+#  EventBridge Connection                                                                                                           #
+##########################################################################
+  EventBridgeConnection:
+    Type: "AWS::Events::Connection"
+    Properties:
+      AuthorizationType: BASIC
+      AuthParameters:
+        BasicAuthParameters:
+          Password: " "
+          Username: "sk_test_4eC39HqLyjWDarjtT1zdp7dc"
```

Comment:
What secret is this? Does it belong to your stripe account our someone else? In either case, I'd suggest not using this.
If you want to use string, may be you could share link to stripe documentation and expect users to generate a key, which they could pass in parameters when deploying the app. Alternatively, you can also SecretsManager to generate a random password and integrate with an API Gateway endpoint using Lambda Authorizer?

Reply:
The template does create EventBridge API Connection with SecretesManager. However, the secretes needs to be passed. Please note I have removed the Username and Password from the template and updated the readme file with instruction to where to get them from.

Reply:
Thanks @biswanathmukherjee. Current setup is also fine, however I believe using parameters this would be easiest for the users as they could just pass that parameter i.e. string secret with `sam deploy` rather than having to go to that file and edit it.